### PR TITLE
Audit and fixed library-functions.md #105

### PR DESCRIPTION
**Pull request by japneetsingh035**
Part of #18
Audit and fix library-functions.md

Check List

- [x]  Look for any typos -> Found multiple typos and fixed them.
- [x]  Make sure all Markdown is correctly done. The conversion was done automatically by a tool, so there may be more idiomatic ways to do some things -> Fixed uppercase letter to suit all other heading letters.
- [x]  Check the page in a Desktop browser. Does it look OK? Are there any issues that need to be fixed? -> Looks great!
- [x]  Check the page in a Mobile browser. Does it look OK? Are there any issues that need to be fixed? -> Looks great!
- [x]  Try the page in both Light and Dark mode. Are there any issues that need to be fixed? -> Looks great!
- [x]  Try running the page through Lighthouse: developers.google.com/web/tools/lighthouse. Are there any issues that need to be fixed? -> Multiple warnings but no issue found.
- [x]  Try running the page through Web Hint: webhint.io. Are there any issues that need to be fixed? -> Multiple warnings but no issue found.
- [x]  Tables that are really wide (types.md) could be converted to use SVG or flex/grid. Make wide visual tables fit in the available viewport. Another example is the table in docs/D-Modularity/pointers.md, where we should reduce the vertical padding/margins on tables. The tables are too large, and probably shouldn't be tables at all. -> No table found
- [x]  Prefer Markdown tables vs. raw HTML tags -> No table found
- [x]  Fix colours in tables so it uses global styles vs. inline styles -> No table found
- [x]  Use alt tags for all images. See supercooldesign.co.uk/blog/how-to-write-good-alt-text for suggestions -> No images found
- [x]  Avoid blockquote for indentation, and prefer q or blockquote only when actually quoting. Use Admonitions instead. -> nothing found
- [x] Example: docs/D-Modulatirty/pointers.md: there are places where > Blockquote are being used and Admonitions could replace it. “Note:” Lightbulb info Admonition. -> nothing found
- [x]  Make sure images work in dark and light modes. For example, docs/8-Computations/logic.md Flags image is white background, needs to work in dark. Create a “negative” or SVG so we can do Dark Mode? Can we programmatically create this negative? -> nothing found
- [x]  Change colours for tables (don’t use Yellow). We should use the default colour theme. -> nothing found
- [x]  Avoid using and to do emphasis. -> nothing found
- [x]  Use backticks and all code for memory addresses and such in tables that use things like void, int, function names, memory addresses. -> correct use of bacticks
- [x]  Make sure all code blocks use syntax highlighting -> correct use of syntax highlighting, not for outputs to make it look different.
- [x]  centre images horizontally (esp. when they are small) so they work well at all responsive sizes -> nothing found
- [x]  check sub-script and super-script that they are correctly used in all cases -> nothing found
- [x]  consider including inter-site links to other pages that have related info -> Works fine!
- [x]  all links should have descriptive text for accessibility -> accessibility tested!
- [x]  Make sure that sub-headings (right-hand side) for each document are correct. -> fixed heading errors
- [x]  Improve readability of code comments (i.e., // and /.../) in dark mode (even light mode might need some work) -> added some readable comments
- [x]  In places where ... was used as a fourth heading level, use #### instead. -> nothing found
- [x]  Fix Frontmatter for the page to include proper id, title, slug, etc. -> Added it to the file in another commit.

### CODE REVIEW

**Review by JiaHua-Zou — COMMENTED** [review]
The changes looks good. Since all the other files has frontmatter for id, title and slug. Maybe add them in if possible.

image

**Comment by japneetsingh035** [comment]
> The changes looks good. Since all the other files has frontmatter for id, title and slug. Maybe add them in if possible.
> 
> image

Thanks! @JiaHua-Zou for reviewing, I added the frontmatter.

File: docs/D-Modularity/library-functions.md [hunk 1]
```diff
@@ -139,14 +143,14 @@ The following program outputs a different set of 10 pseudo-random numbers with e
 ```c
 #include <stdlib.h>
 #include <stdio.h>
-#include <time.h> // prototype for time(NULL)
+#include <time.h> // ...prototype for time(NULL)...
```
Comment (humphd) [inline docs/D-Modularity/library-functions.md:146]: Throughout this file, please remove the `... ...` that you have added.  They aren't needed.

File: docs/D-Modularity/library-functions.md [hunk 2]
```diff
@@ -139,14 +143,14 @@ The following program outputs a different set of 10 pseudo-random numbers with e
...
 
 int main(void)
 {
         int i;
 
-        srand(time(NULL)); // will set a unique seed for each run of the program
-        for (i = 0; i < 10 ; i++)
+        srand(time(NULL)); // ...will set a unique seed for each run of the program...
```
Comment (humphd) [inline docs/D-Modularity/library-functions.md:152]: Remove ...

File: docs/D-Modularity/library-functions.md [hunk 3]
```diff
@@ -139,14 +143,14 @@ The following program outputs a different set of 10 pseudo-random numbers with e
...
+        for (i = 0; i < 10 ; i++) // ...iterate with “i” as index...
```
Comment (humphd) [inline docs/D-Modularity/library-functions.md:153]: Remove ...

**Review by humphd — CHANGES_REQUESTED** [review]

File: docs/D-Modularity/library-functions.md (hunk 2 above)
Comment (japneetsingh035) [reply docs/D-Modularity/library-functions.md]: I removed extra ... from comments in the new commit. Thank you for reviews 👍

**Review by humphd — APPROVED** [review]

**Pull request merged**